R/chromote_session.R

```diff
@@ -85,6 +85,16 @@ ChromoteSession <- R6Class(
           private$pixel_ratio <- value$result$value
         })
 
+      # When a target crashes, raise a warning.
+      # The session cannot be closed because no session id is returned by the
+      # Inspector.targetCrashed event
+      p <- p$
+        then(function(value) {
+          self$Inspector$targetCrashed(timeout_ = NULL, wait_ = FALSE, function(value) {
```

Comment:
I think you'll need to wrap this with 

```R
if (!is.null(self$Inspector$targetCrashed)) {
  ...
}
```

Because some versions of Chrome/Chromium don't have the `Inspector.targetCrashed` event.

Reply:
It might also make sense to call `self$mark_closed()` in the callback.

Reply:
Now that I look at it, I see that the Chromote object _does_ have the Inspector domain. This code runs on my computer:

```R
b <- Chromote$new()
b$Inspector$targetCrashed(
  callback_ = function() { message("Inspector.targetCrashed") },
  wait_ = FALSE
)
```

So I think that the only problem in the existing code is that it needs to check for existence of `self$Inspector$targetCrashed` before it tries to call that method.

Reply:
I'm not sure that Chromote has the Inspector domain, I get an error. Here is a reproducible example

```r
# Use this version https://github.com/rstudio/chromote/tree/ba9bffc291be468c48e658f95fa2bc952a1dbff5
remotes::install_github("rstudio/chromote@ba9bff")

# From a fresh R session
library(chromote)
b <- Chromote$new()
b$debug_messages(TRUE)
# First time, we get an error
b$Inspector$targetCrashed(
  callback_ = function() { message("Inspector.targetCrashed") },
  wait_ = FALSE
)
# The second time, there is no error because chromote does not try
# to send Inspector.enable (a callback is already registered)
b$Inspector$targetCrashed(
  callback_ = function() { message("Inspector.targetCrashed") },
  wait_ = FALSE
)
# But Inspector.enable does not still work
b$Inspector$enable()

# If we open a new target
tab <- ChromoteSession$new()
tab$parent$debug_messages(TRUE)
# chromote sends Inspector.enable
tab$Inspector$targetCrashed(
  callback_ = function() { message("Inspector.targetCrashed") },
  wait_ = FALSE
)
```

Reply:
Thanks, I was able to reproduce the behavior you described.